tests/AbstractFixerTestCase.php

```diff
@@ -0,0 +1,658 @@
+        }
+
+        $configurationDefinition = $this->fixer->getConfigurationDefinition();
+
+        static::assertInstanceOf(FixerConfigurationResolverInterface::class, $configurationDefinition);
+
+        foreach ($configurationDefinition->getOptions() as $option) {
+            static::assertInstanceOf(FixerOptionInterface::class, $option);
+            static::assertNotEmpty($option->getDescription());
+
+            static::assertSame(
+                !isset($this->allowedRequiredOptions[$this->fixer->getName()][$option->getName()]),
+                $option->hasDefault(),
+                sprintf(
+                    $option->hasDefault()
+                        ? 'Option `%s` of fixer `%s` is wrongly listed in `$allowedRequiredOptions` structure, as it is not required. If you just changed that option to not be required anymore, please adjust mentioned structure.'
+                        : 'Option `%s` of fixer `%s` shall not be required. If you want to introduce new required option please adjust `$allowedRequiredOptions` structure.',
+                    $option->getName(),
+                    $this->fixer->getName(),
+                ),
+            );
+
+            static::assertStringNotContainsString(
+                'DEPRECATED',
+                $option->getDescription(),
+                'Option description cannot contain word "DEPRECATED"',
+            );
+        }
+    }
+
+    final public function testFixersReturnTypes()
+    {
+        $tokens = Tokens::fromCode('<?php ');
+        $emptyTokens = new Tokens();
+
+        static::assertIsInt(
+            $this->fixer->getPriority(),
+            sprintf('Return type for ::getPriority of "%s" is invalid.', $this->fixer->getName()),
+        );
+        static::assertIsBool(
+            $this->fixer->supports(new SplFileInfo(__FILE__)),
+            sprintf('Return type for ::supports of "%s" is invalid.', $this->fixer->getName()),
+        );
+
+        static::assertIsBool(
+            $this->fixer->isCandidate($emptyTokens),
+            sprintf('Return type for ::isCandidate with empty tokens of "%s" is invalid.', $this->fixer->getName()),
+        );
+        static::assertFalse($emptyTokens->isChanged());
+
+        static::assertIsBool(
+            $this->fixer->isCandidate($tokens),
+            sprintf('Return type for ::isCandidate of "%s" is invalid.', $this->fixer->getName()),
+        );
+        static::assertFalse($tokens->isChanged());
+
+        if ($this->fixer instanceof HeaderCommentFixer) {
+            $this->fixer->configure(['header' => 'a']);
+        }
+
+        static::assertNull(
+            $this->fixer->fix(new SplFileInfo(__FILE__), $emptyTokens),
+            sprintf('Return type for ::fix with empty tokens of "%s" is invalid.', $this->fixer->getName()),
+        );
+        static::assertFalse($emptyTokens->isChanged());
+
+        static::assertNull(
+            $this->fixer->fix(new SplFileInfo(__FILE__), $tokens),
```

Comment:
assert is wrong, return type of `\PhpCsFixer\Fixer\FixerInterface::fix` is void, better assert that exception was not raised or that some changes occurred.

Reply:
Will be fixed after refactoring